### PR TITLE
Extends CSV by complex types (oneof, map, repeated)

### PR DESCRIPTION
**Pull request by mariuszbarczak**
Signed-off-by: Mariusz Barczak <mariusz.barczak@intel.com>

### CODE REVIEW

File: source/octf/utils/table/Table.cpp [hunk 1]
```diff
@@ -100,33 +102,221 @@ std::ostream &operator<<(std::ostream &os, const Table &table) {
...
+                             const std::string &name) {
+    using namespace google::protobuf;
+
+    bool isRepeated = field->is_repeated();
+
+    switch (field->cpp_type()) {
+    case FieldDescriptor::CppType::CPPTYPE_INT32: {
+        if (isRepeated) {
+            streamRepeatedFieldToRow<int32_t>(row, msg, reflection, field,
+                                              name);
```
Comment (tomaszrybicki) [inline source/octf/utils/table/Table.cpp]: How are we handling variable-sized repeated fields?

File: source/octf/utils/table/Table.cpp [hunk 2]
```diff
@@ -141,87 +331,102 @@ static void streamMessageToRow(Row &row,
...
-        } else {
-            switch (field->cpp_type()) {
-            case FieldDescriptor::CppType::CPPTYPE_INT32: {
-                auto value = reflection->GetInt32(msg, field);
-                row[name] = value;
-                break;
+            if (oneofField && oneofField == field) {
+                std::string oneofName = prefix + oneofDesc->name();
+                std::string oneofFieldName = prefix + oneofField->name();
+                row[oneofName] = oneofFieldName;
```
Comment (tomaszrybicki) [inline source/octf/utils/table/Table.cpp]: Please add some short comments in code

File: source/octf/utils/table/Table.cpp [hunk 3]
```diff
@@ -141,87 +331,102 @@ static void streamMessageToRow(Row &row,
...
+        auto field = desc->field(i);
 
-            case FieldDescriptor::CppType::CPPTYPE_BOOL: {
-                auto value = reflection->GetBool(msg, field);
-                row[name] = value ? "true" : "false";
-                break;
-            }
+        if (field->is_repeated()) {
+            continue;  // Not supported
+        }
```
Comment (tomaszrybicki) [inline source/octf/utils/table/Table.cpp]: Aren't we supporting repeated fields in streamFieldToRow ?

File: source/octf/utils/table/Table.cpp [hunk 4]
```diff
@@ -141,87 +331,102 @@ static void streamMessageToRow(Row &row,
...
+    TableMap map;
+
+    streamMessageToRow(map.getRow(0), *msg, "");
+    streamDescriptionToRow(map.getRow(0), msg->GetDescriptor(), "");
+
+    auto iter = map.getColumnsAssociation().begin();
+    auto end = map.getColumnsAssociation().end();
+    for (; iter != end; iter++) {
+        row[iter->first] = iter->first;
+    }
```
Comment (tomaszrybicki) [inline source/octf/utils/table/Table.cpp]: Please also add comments here

File: source/octf/utils/table/Table.h [hunk 5]
```diff
@@ -219,7 +151,8 @@ Row &operator<<(Row &row, const ::google::protobuf::Descriptor *desc);
  * bar.set_name("Star");
  *
  * // Fill table's header (row 0)
- * table[0] << foo.GetDescriptor() << bar.GetDescriptor();
+ * table::setHeader(table[0], foo);
+ * table::setHeader(table[0], bar);
```
Comment (tomaszrybicki) [inline source/octf/utils/table/Table.h]: Is the intended use to call setHeader with each added (streamed) message?

File: source/octf/utils/table/Table.cpp [hunk 6]
```diff
@@ -141,87 +331,102 @@ static void streamMessageToRow(Row &row,
...
+    streamMessageToRow(row, msg, "");
+    return row;
+}
 
-            case FieldDescriptor::CppType::CPPTYPE_DOUBLE: {
-                auto value = reflection->GetDouble(msg, field);
-                row[name] = value;
-                break;
-            }
+static void streamDescriptionToRow(Row &row,
```
Comment (tomaszrybicki) [inline source/octf/utils/table/Table.cpp]: As we no longer stream Descriptor (using << operator), maybe we should call this setupRowAsHeader or something like that?

File: source/octf/utils/table/Table.h (hunk 5 above)
Comment (mariuszbarczak) [reply source/octf/utils/table/Table.h]: Those two messages are streamed into table, so we need to build a header using them both.

**Pull request merged**